_pages/dev/ssr/server-side-rendering-contextual-logging.md

```diff
@@ -252,3 +252,100 @@ You can also customize the behavior of the `LoggerService` by providing your own
+
+```json
+{
+  "message": "Traceparent header has invalid length: 18. Expected 55 characters.",
+  "context": {
+    "timestamp": "2023-09-13T12:14:21.377Z",
+    "request": {
+      "url": "/electronics-spa/en/USD/"
+    }
+  }
+}
+```
+
+Depending on the error ocurred during the parser, following error messages may be generated:
+- `Traceparent header has invalid length: ${traceparent.length}. Expected 55 characters.` - the `traceparent` header has invalid length
+- `Traceparent header has invalid format.` - the `traceparent` header has invalid format. To learn more, see [W3C Trace Context - Traceparent Header](https://www.w3.org/TR/trace-context/#traceparent-header)
+
+### Connecting Logs to Distributed Traces
+
+**Note:** These steps have been verified on SAP Commerce Cloud, where OpenSearch Dashboards is the default monitoring tool. Please be aware that the steps described here may vary when using other monitoring tools.
+
+All logs generated by Spartacus can be seen in OpenSearch Dashboards monitoring tool.
+The following is an example of a log presented the monitoring tool:
+
+![Log with Trace Context](assets/images/../../../../../assets/images/ssr/contextual_logging_trace_context.png)
+
+OpenSearch enables the formatting of fields as hyperlinks, and this section outlines the steps for configuring links that direct you to the Dynatrace distributed trace view.
+
+To begin, navigate to the main menu and proceed to the "Stack Management" section. Within "Stack Management," choose "Index Patterns" from the available left navigation options.
+
+![Stack Management Path](assets/images/../../../../../assets/images/ssr/contextual_logging_stack_management.png)
+
+Next, select the pre-defined index pattern that corresponds to the log messages generated by your application. Locate the index where your application's logs are displayed.
+
+![Available Indices](assets/images/../../../../../assets/images/ssr/contextual_logging_indices.png)
+
+Once we selected an index pattern, the corresponding field list is shown (see screenshot below). We can filter the fields by name. We use this functionality to search for the field “traceId”. If the field is not found, try refreshing the index pattern with the little reload button on the top right in the screenshot. We can edit the field configuration by clicking on the pencil symbol in the table.
+
+![Index with Logs From Spartacus App](assets/images/../../../../../assets/images/ssr/contextual_logging_index_patterns.png)
+
+After selecting an index pattern, the associated field list will be displayed (as shown in the screenshot below). You can filter the fields by name, and in this case, you should use this feature to search for the "traceId" field. If you can't find the "traceId" field, try refreshing the index pattern by clicking the small reload button located at the top right corner in the screenshot. To edit the field configuration, simply click on the pencil symbol within the table.
```

Comment:
I have problem to understand why the 2nd paragraph repeats what was said in the 1st paragraph. Moreover the 2nd paragraph doesn't seem to describe its relevant image below.